spec/prototype_build_details_comment_action_spec.rb

```diff
@@ -0,0 +1,173 @@
+    end
+
+    it 'includes the direct link if one is provided' do
+      comment = run_described_fastlane_action(
+        appcenter_app_name: 'MyApp',
+        appcenter_release_id: 1337,
+        download_url: 'https://foo.cloudfront.net/someuuid/myapp-prototype-build-pr1337-a1b2c3f.apk'
+      )
+      expect(comment).to include "<td rowspan='4'>"
+      expect(comment).to include "<td><b>Direct Link</b></td><td><a href='https://foo.cloudfront.net/someuuid/myapp-prototype-build-pr1337-a1b2c3f.apk'><tt>myapp-prototype-build-pr1337-a1b2c3f.apk</tt></a></td>"
+    end
+
+    it 'includes the default footnote by default' do
+      comment = run_described_fastlane_action(
+        appcenter_org_name: 'BestOrg',
+        appcenter_app_name: 'MyApp',
+        appcenter_release_id: 1337
+      )
+      expect(comment).to include '<em>Automatticians: You can use our internal self-serve MC tool to give yourself access to App Center if needed.</em>'
+    end
+
+    it 'includes the provided footnote if any' do
+      comment = run_described_fastlane_action(
+        appcenter_app_name: 'MyApp',
+        appcenter_release_id: 1337,
+        footnote: '<em>Note that Google Sign-In in not available in those builds</em>'
+      )
+      expect(comment).to include '<em>Note that Google Sign-In in not available in those builds</em>'
+    end
+  end
+
+  describe 'full comment' do
+    it 'generates a standard HTML table comment by default, with all the information provided' do
+      metadata = {
+        'Version:Short': '28.2',
+        'Version:Long': '28.2.0.108',
+        Flavor: 'Celray'
+      }
+
+      comment = run_described_fastlane_action(
+        appcenter_org_name: 'BestOrg',
+        appcenter_app_name: 'BestApp',
+        appcenter_release_id: 8888,
+        metadata: metadata,
+        footnote: '<em>Note: Google Sign-In in not available in those builds</em>'
+      )
+
+      expect(comment).to eq <<~EXPECTED_COMMENT
+        <p>📲 You can test the changes from this Pull Request in <strong>BestApp</strong> by scanning the QR code below to install the corresponding build via App Center.</p>
+        <table>
+        <tr>
+          <td rowspan='6'><img src='https://chart.googleapis.com/chart?chs=500x500&cht=qr&chl=https%3A%2F%2Finstall.appcenter.ms%2Forgs%2FBestOrg%2Fapps%2FBestApp%2Freleases%2F8888&choe=UTF-8' width='250' height='250' /></td>
+          <td width='150px'><b>App Name</b></td><td><tt>BestApp</tt></td>
+        </tr>
+        <tr><td><b>Version:Short</b></td><td>28.2</td></tr>
+        <tr><td><b>Version:Long</b></td><td>28.2.0.108</td></tr>
+        <tr><td><b>Flavor</b></td><td>Celray</td></tr>
+        <tr><td><b>App Center Build</b></td><td><a href='https://install.appcenter.ms/orgs/BestOrg/apps/BestApp/releases/8888'>Build \#8888</a></td></tr>
+        <tr><td><b>Commit</b></td><td><tt>a1b2c3f</tt></td></tr>
+        </table>
+        <em>Note: Google Sign-In in not available in those builds</em>
+      EXPECTED_COMMENT
```

Comment:
FYI, this is how that example looks like once rendered:<hr/>

<p>📲 You can test the changes from this Pull Request in <strong>BestApp</strong> by scanning the QR code below to install the corresponding build via App Center.</p>
<table>
<tr>
  <td rowspan='6'><img src='https://chart.googleapis.com/chart?chs=500x500&cht=qr&chl=https%3A%2F%2Finstall.appcenter.ms%2Forgs%2FBestOrg%2Fapps%2FBestApp%2Freleases%2F8888&choe=UTF-8' width='250' height='250' /></td>
  <td width='150px'><b>App Name</b></td><td><tt>BestApp</tt></td>
</tr>
<tr><td><b>Version:Short</b></td><td>28.2</td></tr>
<tr><td><b>Version:Long</b></td><td>28.2.0.108</td></tr>
<tr><td><b>Flavor</b></td><td>Celray</td></tr>
<tr><td><b>App Center Build</b></td><td><a href='https://install.appcenter.ms/orgs/BestOrg/apps/BestApp/releases/8888'>Build #8888</a></td></tr>
<tr><td><b>Commit</b></td><td><tt>a1b2c3f</tt></td></tr>
</table>
<em>Note: Google Sign-In in not available in those builds</em>